olive/common/user_module_loader.py

```diff
@@ -16,11 +16,11 @@ class UserModuleLoader:
     Only used for objects that are not json serializable.
     """
 
-    def __init__(self, user_script: Optional[Union[Path, str]], script_dir: Optional[Union[Path, str]] = None):
-        self.user_script = user_script
-        self.script_dir = script_dir
+    def __init__(self, user_script: OLIVE_RESOURCE_ANNOTATIONS, script_dir: OLIVE_RESOURCE_ANNOTATIONS = None):
```

Comment:
what's the scenario that we need the user_script is in AzureML? I am supposing it should be only local file.

Reply:
It is modified to align the dependencies interface. Then Olive can consume all the path dependencies with ResourcePath. Also if user have to mound their code which is hosted in AML, they can directly pass the url.